### PR TITLE
Email sending branch

### PR DESCRIPTION
**Pull request by OksanaDevimaSolutions**

### CODE REVIEW

File: pages/api/register.js [hunk 1]
```diff
@@ -36,13 +37,21 @@ const handler = nc({
...
       // Create token
-      const token = await tokenService.createToken(user.id, email);
+      // const token = await tokenService.createToken(user.id, email);
+
+      // return res.status(201).json({ message: 'user registered!', token });
 
-      return res.status(201).json({ message: 'user registered!', token });
+      return res.status(201).json({
+        message: `user ${user.id} registered! Please confirm your email to login`,
+        uniqueString: user.uniqueString,
```
Comment (dmytro-devima-solutions) [inline pages/api/register.js]: If you sent a secret string back to the client they will be able to call the `verify` endpoint without accessing their inbox.

File: pages/api/verify/[uniqueString].js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import nc from 'next-connect';
+import userService from '../../../src/server/services/user.service';
+
+const handler = nc({
+})
+  .get(async (req, res) => {
+    try {
+      const { uniqueString } = req.query;
+      const user = await userService.findUniqueString(uniqueString);
```
Comment (dmytro-devima-solutions) [inline pages/api/verify/[uniqueString].js]: There would be enough to have a single service method to find and confirm email. No need in calling `findUniqueString` and `addConfirmation` separately.

File: src/server/repositories/user.repository.js [hunk 3]
```diff
@@ -91,6 +92,27 @@ export const findEmail = async (email) => {
 
   return null;
 };
+export const findUniqueString = async (uniqueString) => {
```
Comment (dmytro-devima-solutions) [inline src/server/repositories/user.repository.js]: `findByUniqueString` would make more sense as a name

File: src/server/services/email.service.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import nodemailer from 'nodemailer';
+// import google from 'googleapis';
+const { google } = require('googleapis');
```
Comment (dmytro-devima-solutions) [inline src/server/services/email.service.js]: use `import` instead of `require`.

File: src/server/services/email.service.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+export const sendEmail = (email, uniqueString) => {
+  const { OAuth2 } = google.auth;
+
+  const myOAuth2Client = new OAuth2(process.env.EMAIL_CLIENT_ID, process.env.EMAIL_CLIENT_SECRET, 'https://developers.google.com/oauthplayground');
+
+  myOAuth2Client.setCredentials({ refresh_token: process.env.EMAIL_CLIENT_REFRESH_TOKEN });
+
+  const myAccessToken = myOAuth2Client.getAccessToken();
+  const transporter = nodemailer.createTransport({
```
Comment (dmytro-devima-solutions) [inline src/server/services/email.service.js]: You don't need to create transport for each email. move the transporter outside this method and use it to send emails.

File: pages/api/register.js (hunk 1 above)
Comment (OksanaDevimaSolutions) [reply pages/api/register.js]: yes, obviously. I did it to debug.

**Pull request merged**